src/logic/wallets/store/actions/fetchProvider.ts

```diff
@@ -7,11 +7,12 @@ import { NOTIFICATIONS, enhanceSnackbarForAction } from 'src/logic/notifications
 import enqueueSnackbar from 'src/logic/notifications/store/actions/enqueueSnackbar'
 import { ETHEREUM_NETWORK, ETHEREUM_NETWORK_IDS, getProviderInfo, getWeb3 } from 'src/logic/wallets/getWeb3'
 import { makeProvider } from 'src/logic/wallets/store/model/provider'
+import { updateStoredTransactionsStatus } from 'src/routes/safe/store/actions/transactions/utils/transactionHelpers'
 
 export const processProviderResponse = (dispatch, provider) => {
   const walletRecord = makeProvider(provider)
-
   dispatch(addProvider(walletRecord))
+  updateStoredTransactionsStatus(dispatch, walletRecord)
```

Comment:
Why do we need this? This is a pretty unexpected side effect of connecting a provider :/

Reply:
Yes, it was a last-time fix (that caused some issues while connecting the wallet).

I needed to update the pending txs statuses when switching users (e.g.: `awaiting_your_confirmation` vs `awaiting_confirmations`).

Then I thought that we can implement an observable pattern, just to listen for this kind of changes. But couldn't come out with a quick solution.

BTW, the reasoning behind this is to have the transactions' state mostly persisted in the store, and not inferred in the _presentation layer_.

Reply:
This can be solved via component/selector approach I've written in other comments